services/integrations/leetcode/fetchUserContributions.js

```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+
+const leetcodeGraphQLEndpoint = "https://leetcode.com/graphql/"; // Replace with LeetCode's endpoint if they have one
+
+const fetchData = async (username) => {
+  const graphqlData = {
+    query:
+      "\n  query userProfileCalendar($username: String!, $year: Int) {\n matchedUser(username: $username) {\n  userCalendar(year: $year) {\n   activeYears\n   streak\n   totalActiveDays\n   dccBadges {\n    timestamp\n    badge {\n     name\n     icon\n    }\n   }\n   submissionCalendar\n  }\n }\n}\n  ",
+    variables: {
+      username: username,
+    },
+    operationName: "userProfileCalendar",
+  };
+  const response = await axios.post(leetcodeGraphQLEndpoint, graphqlData);
+  const data = await response.data;
+  const calendar = JSON.parse(
+    data.data.matchedUser.userCalendar.submissionCalendar,
+  );
+
+  const mordified = new Array();
```

Comment:
nameing?? mordified?

Reply:
i was playing around with the api not thinking of naming 😂, sure will change it soon